genref/config.yaml

```diff
@@ -0,0 +1,147 @@
+  - match: ^k8s\.io/apimachinery/pkg/runtime\.Unknown$
+    target: https://godoc.org/k8s.io/apimachinery/pkg/runtime#Unknown
+  - match: ^time\.Duration$
+    target: https://godoc.org/time#Duration
+  - match: ^k8s\.io/(api|apimachinery/pkg/apis)/
+    target: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#{{- lower .TypeIdentifier -}}-{{- arrIndex .PackageSegments -1 -}}-{{- arrIndex .PackageSegments -2 -}}
+
+hideTypePatterns:
+  - "ParseError$"
+  # - "List$"
+
+markdownDisabled: false
+
+stripPrefix:
+  - k8s.io/api/
+  - k8s.io/apimachinery/pkg/apis/
+
+apis:
+  - name: kubelet-config
+    package: k8s.io/kubelet
+    path: config/v1beta1
+    includes:
+      - k8s.io/component-base/config/v1alpha1
+
+  - name: kubelet-stats
+    package: k8s.io/kubelet
+    path: pkg/apis/stats/v1alpha1
+    # This is skipped because the 'stats' API is not yet available in 0.19.3
+    skip: true
+
+  - name: kube-scheduler-config
+    package: k8s.io/kube-scheduler
+    path: config/v1
+
+  - name: kube-scheduler-config
+    package: k8s.io/kube-scheduler
+    path: config/v1beta1
+
+  - name: kube-controller-manager-config
+    package: k8s.io/kube-controller-manager
+    path: config/v1alpha1
+
+  # TODO: This is shared between different controllers so it should be
+  # an includes entry rather than a standalone package.
+  - name: controller-manager-common
+    package: k8s.io/controller-manager
+    path: config/v1alpha1
+
+  - name: kube-proxy-config
+    package: k8s.io/kube-proxy
+    path: config/v1alpha1
+
+  - name: metrics
+    package: k8s.io/metrics
+    path: pkg/apis/metrics/v1alpha1
+
+  - name: metrics
+    package: k8s.io/metrics
+    path: pkg/apis/metrics/v1beta1
+
+  - name: clientauthentication
+    package: k8s.io/client-go
+    path: pkg/apis/clientauthentication/v1alpha1
+
+  - name: clientauthentication
+    package: k8s.io/client-go
+    path: pkg/apis/clientauthentication/v1beta1
+
+  - name: apiserver-config
+    package: k8s.io/apiserver
+    path: pkg/apis/apiserver/v1alpha1
+
+  # This only exists in v0.19.3
+  - name: apiserver-config
+    package: k8s.io/apiserver
+    path: pkg/apis/apiserver/v1beta1
+
+  - name: apiserver-config
+    package: k8s.io/apiserver
+    path: pkg/apis/apiserver/v1
+
+  - name: apiserver-encryption
+    package: k8s.io/apiserver
+    path: pkg/apis/config/v1
+
+  - name: apiserver-audit
+    package: k8s.io/apiserver
+    path: pkg/apis/audit/v1
+
+  - name: apiserver-webhookadmission
+    package: k8s.io/apiserver
+    path: pkg/admission/plugin/webhook/config/apis/webhookadmission/v1alpha1
+
+  - name: apiserver-webhookadmission
+    package: k8s.io/apiserver
+    path: pkg/admission/plugin/webhook/config/apis/webhookadmission/v1
+
+  - name: apiserver-eventratelimit
+    package: k8s.io/kubernetes
+    path: plugin/pkg/admission/eventratelimit/apis/eventratelimit/v1alpha1
+    # Skipped because k8s.io/kubernetes cannot be imported
+    skip: true
```

Comment:
Is this a configuration that the tool requires or is this configuration a custom configuration?
If custom you may want to create a sub-directory for specific types of output?

Reply:
This is a config required by the tool. It contains some default packages and paths for the tool to generate reference from. Users can customize this config to add new packages and paths.

As for output, we only support a custom output path. Since each package has its own package name and version number, there should be no conflict in the output file names.

Beyond that, we allow users to specify `-include` and `-exclude` when running the tool. So users can easily specify which output goes to which path.